homeassistant/components/homekit/accessories.py

```diff
@@ -228,8 +234,12 @@ def get_accessory(  # noqa: C901
             a_type = "LightSensor"
 
     elif state.domain == "switch":
-        switch_type = config.get(CONF_TYPE, TYPE_SWITCH)
-        a_type = SWITCH_TYPES[switch_type]
+        if switch_type := config.get(CONF_TYPE):
+            a_type = SWITCH_TYPES[switch_type]
+        elif state.attributes.get(ATTR_DEVICE_CLASS) == SwitchDeviceClass.OUTLET:
+            a_type = "Outlet"
+        else:
+            a_type = "Switch"
```

Comment:
How come this is changed? Should it be in another PR?

Reply:
I wrote the test for this and than I realized it didn't work because the check for the device class was missing.

I'll break it out into another PR

Reply:
https://github.com/home-assistant/core/pull/101760